src/components/datepicker/datePicker.js

```diff
@@ -179,11 +202,23 @@
     this.ngModelCtrl = null;
 
     /** @type {HTMLInputElement} */
-    this.inputElement = $element[0].querySelector('input');
+    this.inputElement = $element[0].querySelector('input.begin');
 
     /** @final {!angular.JQLite} */
     this.ngInputElement = angular.element(this.inputElement);
 
+    /** @type {HTMLInputElement} */
+    this.beginInputElement = $element[0].querySelector('input.begin');
+
+    /** @final {!angular.JQLite} */
+    this.ngBeginInputElement = angular.element(this.beginInputElement);
```

Comment:
The HTMLInputElement does have native impl's for oninput and onkeydown (which is what the wrapped elements are being used for), however the function serving as the event handler does not have the event passed as a param. This event param is needed by the functions currently being passed in. I think this is why this approach was initially used.  e.g. --> http://www.w3schools.com/jsref/event_onkeydown.asp